edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/ScopeIdentitiesHandler.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Identity.Service;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public class ScopeIdentitiesHandler : IMessageProducer
+    {
+        const string AddOrUpdateTopic = "$internal/identities/addOrUpdate";
+        const string RemoveTopic = "$internal/identities/removeTopic";
+        readonly IDeviceScopeIdentitiesCache deviceScopeIdentitiesCache;
+        readonly AtomicBoolean connected = new AtomicBoolean(false);
+        IMqttBrokerConnector connector;
+        Dictionary<string, BrokerServiceIdentity> brokerServiceIdentities = new Dictionary<string, BrokerServiceIdentity>();
+
+        public ScopeIdentitiesHandler(IDeviceScopeIdentitiesCache deviceScopeIdentitiesCache)
+        {
+            this.deviceScopeIdentitiesCache = deviceScopeIdentitiesCache;
+            this.deviceScopeIdentitiesCache.ServiceIdentityUpdated += (sender, serviceIdentity) => Task.WhenAll(this.ServiceIdentityUpdated(serviceIdentity));
+            this.deviceScopeIdentitiesCache.ServiceIdentityRemoved += (sender, identity) => Task.WhenAll(this.ServiceIdentityRemoved(identity));
```

Comment:
this is the race concern that I brought up this morning.  The refresh starts when the cache object is instantiated, and an indeterminable amount of time has passed between and you hooking up the delegates here.  There's no guarantee that you won't be missing callbacks.